### PR TITLE
Remove Dockerfile from contributing + Add citations

### PR DESCRIPTION
**Pull request by SusiJo**
The Dockerfile in `CONTRIBUTING.md` pipeline template is outdated and there were some citations missing and needed formatting.

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

**Comment by SusiJo** [comment]
Not sure though if the `[Dockerfile](.devcontainer/Dockerfile)` line also needs to be removed in the root `.github/CONTRIBUTING.md`.

**Comment by mashehu** [comment]
Yes, please remove it there as well

File: nf_core/pipeline-template/CITATIONS.md
Comment (mashehu) [inline nf_core/pipeline-template/CITATIONS.md:1]: Was the formatting wrong here?

**Review by mashehu — APPROVED** [review]

File: nf_core/pipeline-template/CITATIONS.md
Comment (SusiJo) [reply nf_core/pipeline-template/CITATIONS.md:1]: Yes, 2 citations were missing and some blank lines in between the tool and the citation.

**Pull request merged**